app/models/container_template.rb

```diff
@@ -12,4 +12,37 @@ class ContainerTemplate < ApplicationRecord
   serialize :objects, Array
 
   acts_as_miq_taggable
+
+  def instantiate(params, project = nil)
+    project ||= container_project.name
+    processed_template = process_template(ext_management_system.connect,
+                                          :metadata   => {
+                                            :name      => name,
+                                            :namespace => project
+                                          },
+                                          :objects    => objects,
+                                          :parameters => params
+                                         )
+    create_objects(processed_template['objects'], project) if processed_template
+  end
+
+  def process_template(client, template)
+    client.process_template(template)
+  rescue KubeException => e
```

Comment:
We normally rescue provider exceptions and re-raise a corresponding miq-exception.